traitsui/examples/demo/Standard_Editors/tests/test_ButtonEditor_demo.py

```diff
@@ -0,0 +1,68 @@
+"""
+This example demonstrates how to test interacting with a button created
+using ButtonEditor.
+It also demonstarates the use of UI Tester, together with pyface's
+ModalDialogTester.
+
+The GUI being tested is written in the demo under the same name (minus the
+preceding 'test') in the outer directory.
+"""
+
+import os
+import runpy
+import unittest
+
+# FIXME: Import from api instead
+# enthought/traitsui#1173
+from traitsui.testing.tester import command
+from traitsui.testing.tester.ui_tester import UITester
+
+from traitsui.tests._tools import (
+    requires_toolkit,
+    ToolkitName,
+)
```

Comment:
Demo examples are where we point developers to when we say "here is how you do XYZ". So we should expect this code being copied and pasted, then adapted by external projects.
Importing from the private `_tools` is not recommended, so we would need to handle the absence of `ModalDialogTester` on wx differently.

This might be what we can do here instead:
https://github.com/enthought/pyface/blob/4fda22878e64d930ce0d2bebac69907ea71b31b9/pyface/tests/test_dialog.py#L26-L29

Then skip the test if `no_modal_dialog_tester` is true.
https://github.com/enthought/pyface/blob/4fda22878e64d930ce0d2bebac69907ea71b31b9/pyface/tests/test_dialog.py#L109

Reply:
> Importing from the private _tools is not recommended

It is okay for traitsui to import traitsui's private module, so it is okay this test was in traitsui test suite not designated as demo examples.